src/backends/sentry_backend_crashpad.cpp

```diff
@@ -434,23 +434,21 @@ crashpad_backend_startup(
     // `sentry_init` will persist the upload flag.
     data->db = crashpad::CrashReportDatabase::Initialize(database).release();
     data->client = new crashpad::CrashpadClient;
-    bool success;
     char *minidump_url
         = sentry__dsn_get_minidump_url(options->dsn, options->user_agent);
     if (minidump_url) {
         SENTRY_TRACEF("using minidump URL \"%s\"", minidump_url);
-        success = data->client->StartHandler(handler, database, database,
-            minidump_url, options->http_proxy ? options->http_proxy : "",
-            annotations, arguments,
-            /* restartable */ true,
-            /* asynchronous_start */ false, attachments);
-        sentry_free(minidump_url);
     } else {
         SENTRY_WARN(
             "failed to construct minidump URL (check DSN or user-agent)");
-        crashpad_state_dtor(data);
-        return 1;
     }
+    const std::string url
+        = minidump_url ? std::string(minidump_url) : std::string();
+    sentry_free(minidump_url);
+    bool success = data->client->StartHandler(handler, database, database, url,
+        options->http_proxy ? options->http_proxy : "", annotations, arguments,
+        /* restartable */ true,
+        /* asynchronous_start */ false, attachments);
```

Comment:
Why don't we do it like we do it for `http_proxy` without explicitly wrapping it in `std::string`? A wrap will happen when passing the argument anyway.

```suggestion
    bool success = data->client->StartHandler(handler, database, database, 
        minidump_url ? mindump_url : "",
        options->http_proxy ? options->http_proxy : "", annotations, arguments,
        /* restartable */ true,
        /* asynchronous_start */ false, attachments);
    sentry_free(minidump_url);
```